README.md

```diff
@@ -177,63 +177,59 @@ Now, the `output_tabulator()` function can be called anywhere we want to render
-    _meta: TransformerMetadata,
-    _fn: ValueFn[pd.DataFrame | None],
-):
-    res = await resolve_value_fn(_fn)
-    if res is None:
-        return None
-
-    if not isinstance(res, pd.DataFrame):
-        # Throw an error if the value is not a dataframe
-        raise TypeError(f"Expected a pandas.DataFrame, got {type(res)}. ")
-
-    # Get data from dataframe as a list of lists where each inner list is a
-    # row, column names as array of strings and types of each column as an
-    # array of strings
-    return {
-        "data": res.values.tolist(),
-        "columns": res.columns.tolist(),
-        "type_hints": res.dtypes.astype(str).tolist(),
-    }
+from shiny.render.renderer import Jsonifiable, Renderer
+
+
+class render_tabulator(Renderer[pd.DataFrame]):
+    """
+    Render a pandas dataframe as a tabulator table.
+    """
+
+    def auto_output_ui(self):
+        """
+        Express UI for the tabulator renderer
+        """
+        return ui.output_tabulator(self.output_name)
+
+    async def transform(self, value: pd.DataFrame) -> Jsonifiable:
+        """
+        Transform a pandas dataframe into a JSONifiable object that can be
+        passed to the tabulator HTML dependency.
+        """
+        if not isinstance(value, pd.DataFrame):
+            # Throw an error if the value is not a dataframe
+            raise TypeError(f"Expected a pandas.DataFrame, got {type(value)}. ")
+
+        # Get data from dataframe as a list of lists where each inner list is a
+        # row, column names as array of strings and types of each column as an
+        # array of strings
+        return {
+            "data": value.values.tolist(),
+            "columns": value.columns.tolist(),
+            "type_hints": value.dtypes.astype(str).tolist(),
+        }
 ```
 
-The `output_transformer` decorator is a decorator factory that takes a function that returns a dictionary of data to be passed to the client side. The function that it decorates is passed two arguments: `_meta` and `_fn`.
+The `Renderer[IT]` class is a decorator that renders output value functions into Jsonifiable objects. A renderer subclass is composed of three main parts: `IT`, `auto_output_ui` and either a `transform` or `render` function.
 
-`_meta` is a dictionary of metadata about the function that is being decorated. We don't use it in our example.
+`IT` represents the type of the value returned by an output value function. It helps users with IDE support know what kind of objects may be returned by their output value function.
 
-`_fn` is the function that is being decorated. Aka the function that goes below the `@render_tabulator()` in your app's server code. In this case we are expecting that that function returns either a pandas dataframe or `None`.
+The `auto_output_ui()` method is used to generate the UI for the output if the renderer were to be used in Express mode. In this case we just use the `output_tabulator()` function we wrote earlier.
```

Comment:
Update this to reflect any changes made to https://github.com/posit-dev/py-shiny-site/pull/68